tests/BuildSystem/Build/discovered-makefile-deps-relative.llbuild

```diff
@@ -0,0 +1,58 @@
+# Check the handling of compiler-based discovered dependencies with relative
+# paths
+
+# We test with a simple command that just writes a fake .d file which lists an
+# input dependency on "header-1".
+
+# RUN: rm -rf %t.build
+# RUN: mkdir -p %t.build/wd
+# RUN: touch %t.build/wd/header-1 %t.build/wd/input-1
+# RUN: cp %s %t.build/build.llbuild
+
+
+# Check the first build.
+#
+# RUN: %{llbuild} buildsystem build --serial --chdir %t.build > %t1.out
+# RUN: %{FileCheck} --check-prefix=CHECK-INITIAL --input-file=%t1.out %s
+#
+# CHECK-INITIAL: CC output-1
+# CHECK-INITIAL: cat output-1 > ../output
+
+
+# Check a build that modifies the header.
+#
+# RUN: echo "mod" >> %t.build/header-1
+# RUN: %{llbuild} buildsystem build --serial --chdir %t.build &> %t2.out
+# RUN: %{FileCheck} --check-prefix=CHECK-AFTER-MOD --input-file=%t2.out %s
+#
+# CHECK-AFTER-MOD: CC output-1
+# CHECK-AFTER-MOD: cat output-1 > ../output
+#
+# REQUIRES: platform=Darwin
```

Comment:
Why does this require Darwin?

Reply:
Its using the working-directory support, which is not yet implemented on linux (requires #400 to be finished).

Reply:
Ah yes, I should deal with that...